scripts/o3de/o3de/repo_properties.py

```diff
@@ -0,0 +1,399 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+"""
+This file contains all the code that has to do with editing the remote repo template
+"""
+import argparse
+import pathlib
+import sys
+import json
+import logging
+import hashlib
+import shutil
+
+from o3de import manifest, utils, validation
+
+logger = logging.getLogger('o3de.repo_properties')
+logging.basicConfig(format=utils.LOG_FORMAT)
+
+
+def merge_json_data(json_path: pathlib.Path, json_data: dict) -> dict:
+    """
+    Merges the json data with the existing json data if it exists.
+    If the json data does not exist, it is created.
+    The json data is saved to the json path.
+    Returns the merged json data.
+    
+    Args:
+        json_path (pathlib.Path): The path to the json file.
+        json_data (dict): The json data to merge with the existing json data.
+    
+    Returns:
+        dict: The merged json data.
+    
+    Raises:
+        FileNotFoundError: If the json path does not exist.
+        ValueError: If the json data is not a dict.
+    """
+    merged_json_data = {}
+    if json_path.exists():
+        existing_json_data = {}
+        with open(json_path, 'r') as f:
+            existing_json_data = json.load(f)
+
+        merged_json_data = existing_json_data | json_data
+
+        if not 'version' in merged_json_data:
+            merged_json_data['version'] = '1.0.0'
+
+        with open(json_path, 'w') as f:
+            f.write(json.dumps(merged_json_data, indent=4) + '\n')
+
+    return merged_json_data
+
+def create_release(src_data_path: pathlib.Path, 
+                   json_data_path: pathlib.Path, 
+                   archive_filename: pathlib.PurePath, 
+                   releases_path: pathlib.Path, 
+                   repo_uri:str,
+                   download_prefix:str) -> dict:
+    """
+    Creates a release for the given src_data_path.
+    The release is saved to the releases_path.
+    Returns the json data.
```

Comment:
Informational question, but what is a "release"?

Reply:
The naming could probably be improved, this name refers to a release of a specific version of a remote object.  So every time you update a `repo.json` with a new version of a gem/project/template and tell us to make an archive, it calls this function and creates a "release", maybe we should instead calls this `create_remote_object_archive`